src/sec_certs/dataset/dataset.py

```diff
@@ -65,7 +65,7 @@ def __init__(
         self.name = name if name else type(self).__name__
         self.description = description if description else datetime.now().strftime("%d/%m/%Y %H:%M:%S")
         self.state = state if state else self.DatasetInternalState()
-        self.root_dir = Path(root_dir)
+        self.root_dir = Path(root_dir) if root_dir is not None else None  # type: ignore
```

Comment:
Why is the `# type: ignore` required? Shouldn't `self.root_dir` expect `Path | None` assignments?